tests/extension/oneapi_memcpy2d/memcpy2d_queue_shortcut.h

```diff
@@ -0,0 +1,250 @@
+                                      src_pitch, region_width, region_height);
+
+        dest_address = dst2.get() + shift_row * dest_pitch + shift_col;
+        src_address = dst1.get() + shift_row * dest_pitch + shift_col;
+        auto queue2 = queue.ext_oneapi_memcpy2d(
+            dest_address, dest_pitch, src_address, dest_pitch, region_width,
+            region_height, queue1);
+
+        dest_address = dst3.get() + shift_row * dest_pitch + shift_col;
+        src_address = dst2.get() + shift_row * dest_pitch + shift_col;
+        queue.ext_oneapi_memcpy2d(dest_address, dest_pitch, src_address,
+                                  dest_pitch, region_width, region_height,
+                                  {queue1, queue2});
+
+        queue.wait();
+        copy_destination_to_host_result<DestPtrType>(dst3.get(), result,
+                                                     result_size, queue);
+
+        for (size_t i = 0; i < array_height; ++i) {
+          for (size_t j = 0; j < dest_pitch; ++j) {
+            size_t address = j + dest_pitch * i;
+            T val = get_expected_value(address, init_v, expected_v);
+            CHECK(val == result[address]);
+          }
+        }
+      }
+    }
+    SECTION(sycl_cts::section_name(std::string("Check copy2d with T = ") +
+                                   t_name +
+                                   " src_ptr_type = " + src_ptr_type_name +
+                                   " and"
+                                   " dest_ptr_type = " +
+                                   dest_ptr_type_name)
+                .create()) {
+      auto dest_address = dst1.get() + (shift_row * dest_pitch + shift_col);
+      auto src_address = src.get() + (shift_row * src_pitch + shift_col);
+      auto queue1 =
+          queue.ext_oneapi_copy2d(src_address, src_pitch, dest_address,
+                                  dest_pitch, region_width, region_height);
+
+      dest_address = dst2.get() + shift_row * dest_pitch + shift_col;
+      src_address = dst1.get() + shift_row * dest_pitch + shift_col;
+      auto queue2 = queue.ext_oneapi_copy2d(
+          src_address, dest_pitch, dest_address, dest_pitch, region_width,
+          region_height, queue1);
+
+      dest_address = dst3.get() + shift_row * dest_pitch + shift_col;
+      src_address = dst2.get() + shift_row * dest_pitch + shift_col;
+      queue.ext_oneapi_copy2d(src_address, dest_pitch, dest_address, dest_pitch,
+                              region_width, region_height, {queue1, queue2});
+
+      queue.wait();
+      copy_destination_to_host_result<DestPtrType>(dst3.get(), result,
+                                                   result_size, queue);
+
+      for (size_t i = 0; i < array_height; ++i) {
+        for (size_t j = 0; j < dest_pitch; ++j) {
+          size_t address = (j + dest_pitch * i);
+          T val = get_expected_value(address, init_v, expected_v);
+          CHECK(val == result[address]);
+        }
+      }
```

Comment:
DRY: all these should resolve to:
```suggestion
      for_index(array_height, dest_pitch, [&] (auto index) {
          T val = get_expected_value(address, init_v, expected_v);
          CHECK(val == result[address]);
      });
```

Reply:
Requested change applied. Thanks for clarifications.